libs/balances-and-allowances/src/state/allowancesAtom.ts

```diff
@@ -1,7 +1,47 @@
+import { atom } from 'jotai'
 import { atomWithReset } from 'jotai/utils'
 
+import { BigNumber } from '@ethersproject/bignumber'
+
+import ms from 'ms.macro'
+
 import { Erc20MulticallState } from '../types'
 
-export interface AllowancesState extends Erc20MulticallState {}
+/**
+ * Priority value is valid for 30 seconds after tx mined
+ * After that, we use allowance from Updaters
+ */
+const PRIORITY_VALUE_TTL = ms`30s`
+
+export interface AllowancesState extends Erc20MulticallState {
+  /**
+   * Since we update allowances periodically, we have a lag between the current allowance (in atom) and the actual allowance (blockchain)
+   * To avoid this, we have a priority value that is updated immediately after tx mined (see FinalizeTxUpdater)
+   */
+  priorityValues: {
+    [address: string]: {
+      value: BigNumber | undefined
```

Comment:
All the other parts I understood, and made sense!

But was really a new state necessary? I mean, you eagerly get the value of the approval once the TX is mined. 

Why is not enough to just update the normal allowance state. Probably, next run for getting all allowances will re-fetch again the value, and will likely match the one we got eagerly

wasn't it simpler without these 2 states and a TTL? 

probably Im missing something

Reply:
Good question!

>Probably, next run for getting all allowances will re-fetch again the value, and will likely match the one we got eagerly

Unfortunately, it doesn't work in 100% of cases. Quite often nodes are not in the same state and the next re-fetch returns the outdated value, I've faced it several times.